modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -939,6 +947,62 @@ Assignment generateAssignment(Experiment experiment, User.ID userID, Context con
         Assignment result = builder.build();
         return assignmentsRepository.assignUser(result, experiment, date);
     }
+
+    Assignment createAssignmentObject(Experiment experiment, User.ID userID, Context context, boolean selectBucket,
```

Comment:
Please add comment to describe what this new method(createAssignmentObject) does. It looks like this method is only used by batch assignment call. Am I correct? It will be good to rename generateAssignment() to generateSingleAssignment() to help reader to understand the code.